docs/apache-airflow-providers-edge/edge_executor.rst

```diff
@@ -0,0 +1,248 @@
+ .. Licensed to the Apache Software Foundation (ASF) under one
+    or more contributor license agreements.  See the NOTICE file
+    distributed with this work for additional information
+    regarding copyright ownership.  The ASF licenses this file
+    to you under the Apache License, Version 2.0 (the
+    "License"); you may not use this file except in compliance
+    with the License.  You may obtain a copy of the License at
+
+ ..   http://www.apache.org/licenses/LICENSE-2.0
+
+ .. Unless required by applicable law or agreed to in writing,
+    software distributed under the License is distributed on an
+    "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+    KIND, either express or implied.  See the License for the
+    specific language governing permissions and limitations
+    under the License.
+
+Edge Executor
+=============
+
+.. note::
+
+    The Edge Provider Package is an experimental preview. Features and stability is limited
+    and needs to be improved over time. Target is to have full support in Airflow 3.
+    Once Airflow 3 support contains Edge Provider, maintenance of the Airflow 2 package will
+    be dis-continued.
+
+
+.. note::
+
+    As of Airflow 2.10.0, you can install the ``edge`` provider package to use this executor.
+    This can be done by installing ``apache-airflow-providers-edge`` or by installing Airflow
+    with the ``edge`` extra: ``pip install 'apache-airflow[edge]'``.
+
+
+``EdgeExecutor`` is an option if you want to distribute tasks to workers distributed in different locations.
+You can use it also in parallel with other executors if needed. Change your ``airflow.cfg`` to point
+the executor parameter to ``EdgeExecutor`` and provide the related settings.
+
+The configuration parameters of the Edge Executor can be found in the Edge provider's :doc:`configurations-ref`.
+
+Here are a few imperative requirements for your workers:
+
+- ``airflow`` needs to be installed, and the CLI needs to be in the path
+- Airflow configuration settings should be homogeneous across the cluster
+- Operators that are executed on the Edge Worker need to have their dependencies
+  met in that context. Please take a look to the respective provider package
+  documentations
+- The worker needs to have access to its ``DAGS_FOLDER``, and you need to
+  synchronize the filesystems by your own means. A common setup would be to
+  store your ``DAGS_FOLDER`` in a Git repository and sync it across machines using
+  Chef, Puppet, Ansible, or whatever you use to configure machines in your
+  environment. If all your boxes have a common mount point, having your
+  pipelines files shared there should work as well
+
+
+Minimum configuration for the Edge Worker to make it running is:
+
+- Section ``[core]``
+
+  - ``executor``: Executor must be set or added to be ``airflow.providers.edge.executors.EdgeExecutor``
```

Comment:
I believe I saw in another PR you added this executor to the holy list of "core" executors. So folks should be able to use the short name instead of the module path right?

Reply:
Yes, for Core this is in https://github.com/apache/airflow/pull/41730

For Breeze it is in https://github.com/apache/airflow/pull/41731

(Separated as we have the rule to split provider and core PRs to cherry-pick)

Reply:
So in that case the value does not need to be the module path, it can just be "EdgeExecutor".

Btw, how did we decide to add another executor to that holy list? I thought we were no longer doing so?

Reply:
Oh, I did not know that the list of executors is "holy" - was there a decision that only specific "premium" Executors shall be added? Maybe I am too new or missed this. So if I "should not" then this would just be affecting #41730. I "naively thought" having the full module path here until the other PR is merged, then with a future Airflow version the short name could be used. But I would not insist on a short name. You/the community is able to have a word on this and convince me of course.

Reply:
Yeah, that list is really only for executors that airflow vends, they're sometimes called "core" executors. Local, Sequential, Debug are all examples. Celery and Kubernetes now live in provider packages, but are grandfathered into being in this holy list because they existed before executors lived in provider packages. Since your executor exists in a provider package, my vote would be to _not_ add it to the holy list in the executor loader and then if that's the case, then users would then need the full module path like you have in the docs now.

Reply:
Okay, then I'll adjust #41730 and revert 99% of the changes there. Then the docs here will stay as it is fr the time being. (Until this one is stable, performant, cool and getting grandfathered :-D)